src/components/atoms/AltinnAttachment.tsx

```diff
@@ -1,138 +1,49 @@
-  listDisableVerticalPadding?: boolean;
-  /** Adds 1.25rem paddingLeft */
-  nested?: boolean;
   id?: string;
+  title?: string;
 }
 
-function ListItemLink(props: any) {
-  return (
-    <ListItem
-      button={true}
-      component='a'
-      {...props}
-    />
-  );
-}
+export function AltinnAttachment({ attachments, id, title }: IAltinnAttachmentProps) {
+  const { lang, langAsString, selectedLanguage } = useLanguage();
 
-export function AltinnAttachment({ attachments, listDisableVerticalPadding, nested, id }: IAltinnAttachmentProps) {
-  const classes = useStyles();
-  const { selectedLanguage } = useLanguage();
+  const filteredAndSortedAttachments = attachments
+    ?.filter((attachment) => attachment.name)
+    .sort((a, b) => (a.name && b.name ? a.name.localeCompare(b.name, selectedLanguage) : 0));
 
   return (
-    <>
-      <List
-        disablePadding={Boolean(listDisableVerticalPadding)}
-        id={id}
-        data-testid='attachment-list'
-      >
-        {attachments &&
-          attachments
-            .sort((a, b) => {
-              if (!a.name || !b.name) {
-                return 0;
-              }
-              return a.name.localeCompare(b.name, selectedLanguage);
-            })
-            .map((attachment, index) => (
-              <ListItemLink
-                className={cn(
-                  {
-                    [classes.listItemPadding]: nested === true,
-                    [classes.listItemPaddingNone]: nested !== true,
-                  },
-                  classes.a,
-                )}
-                href={attachment.url && makeUrlRelativeIfSameDomain(attachment.url)}
-                key={index}
-              >
-                <ListItemIcon>
-                  <FileExtensionIcon
-                    fileEnding={getFileEnding(attachment.name)}
-                    className={classes.icon}
-                  />
-                </ListItemIcon>
-                <ListItemText
-                  primary={
-                    <>
-                      <Typography
-                        variant='body1'
-                        className={cn(classes.inline, classes.primaryText)}
-                      >
-                        {attachment.name}
-                      </Typography>
-                      <Typography
-                        variant='body1'
-                        className={classes.inline}
-                      >
-                        &nbsp;(last ned)
-                      </Typography>
-                    </>
-                  }
-                  classes={{
-                    root: cn(classes.listItemTextPadding),
-                  }}
-                />
-              </ListItemLink>
-            ))}
-      </List>
-    </>
+    <List
+      id={id}
+      heading={lang(title)}
+      data-testid='attachment-list'
+      className={classes.attachmentList}
+    >
+      {filteredAndSortedAttachments &&
```

Comment:
You could use optional chaining here `filteredAndSortedAttachments?.map(...`

Reply:
Will do, thx!